Realm/Tests/MigrationTests.mm

```diff
@@ -307,6 +307,38 @@ - (void)testStringPrimaryKeyNoIndexMigration {
     [RLMRealm migrateRealmAtPath:RLMTestRealmPath()];
 }
 
+- (void)testIntPrimaryKeyNoIndexMigration {
+    // make string an int
+    RLMObjectSchema *objectSchema = [RLMObjectSchema schemaForObjectClass:MigrationPrimaryKeyObject.class];
+
+    // create without search index
+    objectSchema.primaryKeyProperty.attributes = 0;
+
+    // create realm with old schema and populate
+    @autoreleasepool {
+        RLMRealm *realm = [self realmWithSingleObject:objectSchema];
+        [realm beginWriteTransaction];
+        [realm createObject:MigrationPrimaryKeyObject.className withObject:@[@1]];
+        [realm createObject:MigrationPrimaryKeyObject.className withObject:@[@2]];
+        [realm commitWriteTransaction];
+
+        XCTAssertFalse(realm.schema[MigrationPrimaryKeyObject.className].table->has_search_index(0));
```

Comment:
I would prefer the pattern where we have a block and XCTAssertThrows. Every time I see this pattern it takes me a minute to figure out why we XCTAssertFalse which is annoying.